Graph/Topological-Sort.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+  using namespace std;
+
+int n , m; // For number of Vertices (V) and number of edges (E)
+vector< vector<int> > G;
+vector<bool> visited;
+vector<int> ans;
+
+void dfs(int v) {
+  visited[v] = true;
+  for (int u : G[v]) {
+    if (!visited[u])
+      dfs(u);
+  }
+  ans.push_back(v);
+}
+ 
+void topological_sort() {
+  visited.assign(n, false);
+  ans.clear();
+  for (int i = 0; i < n; ++i) {
+    if (!visited[i])
+      dfs(i);
+  }
+  reverse(ans.begin(), ans.end());
+}
+int main(){
+  cin >> n >> m;
+  int x , y;
+  G.resize(n , vector<int>());
+  for(int i = 0 ; i < n ; ++i) {
+    cin >> x >> y;
+    x-- , y--; //to convert 1-indexed to 0-indexed
+    G[x].push_back(y);
+  }
+  topological_sort();
+  cout << "Topological Order : \n";
+  for(int v : ans) {
+    cout << v << ' ';
```

Comment:
if you are _converting 1-indexed to 0-indexed_ after input, you should _convert 0-indexed to 1-indexed_ before output too. _(User will input **1** vertex but will see **0** in output)_

Reply:
Fixed